addons/panku_console/common/config.gd

```diff
@@ -1,25 +1,36 @@
 class_name PankuConfig
 
-const FILE_NAME = "panku_config.cfg"
-const FILE_PATH = "user://" + FILE_NAME
+const USER_CONFIG_FILE_PATH = "user://panku_config.cfg"
 
+# load config from file, always return a dictionary
+static func _get_config(file_path:String) -> Dictionary:
+	if FileAccess.file_exists(file_path):
+		var file = FileAccess.open(file_path, FileAccess.READ)
+		var content := file.get_as_text()
+		var config:Dictionary = str_to_var(content)
+		if config: return config
+	return {}
+
+# save user config to file
 static func set_config(config:Dictionary):
-	var file = FileAccess.open(FILE_PATH, FileAccess.WRITE)
+	var file = FileAccess.open(USER_CONFIG_FILE_PATH, FileAccess.WRITE)
 	var content = var_to_str(config)
 	file.store_string(content)
 
+# get config, if user config exists, return user config, otherwise return default config configured by plugin user
 static func get_config() -> Dictionary:
-	if FileAccess.file_exists(FILE_PATH):
-		var file = FileAccess.open(FILE_PATH, FileAccess.READ)
-		var content = file.get_as_text()
-		var config:Dictionary = str_to_var(content)
-		if config: return config
-	return {}
+	var user_config:Dictionary = _get_config(USER_CONFIG_FILE_PATH)
+	if not user_config.is_empty():
+		return user_config
+	# if no user config, return default config, which is read-only
+	var default_config:Dictionary = _get_config(PankuConsolePlugin.get_custom_default_config_path())
+	print(PankuConsolePlugin.get_custom_default_config_path())
+	return default_config
```

Comment:
The `print` statement spams dozen of lines when `user://` has no config file. Can we rid off this debug info or move it somewhere to call only once?

Also if developer put invalid config path into project settings the function will return empty dictionary, it's not corresponds debug message
>[Panku Console] Default config file not found. Using code-level default config.

The function can be modified to actually return code level default config. Something like
```suggestion
	# if no user config, return default config, which is read-only
	if PankuConsolePlugin.is_custom_default_config_exists():
		return _get_config(PankuConsolePlugin.get_custom_default_config_path())
		
	return _get_config(PankuConsolePlugin.INITIAL_DEFAULT_CONFIG_FILE_PATH)
```

Reply:
Thanks a lot!